### DISCUSSION TITLE
repeater: access complete current item

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#12898] [Help]
### Package

Form builder

### Package Version

v3.2.80

### How can we help you?

I can use `$get` to [access the parent field values](https://filamentphp.com/docs/3.x/forms/fields/repeater#using-get-to-access-parent-field-values), but how can I access the full current item?

E.g. I have as a object structure (all in single JSON):
```
- items
  - [field1, field2, field3, field4, ...]
  - [field1, field2, field3, field4, ...]
```

In my repeater:
- I have form fields for field1 and field2
- I want to show the rest of the fields as array string: `['field3' => ..., 'field4' => ..., ...]`.

I tried something like:
```
Repeater::make('items')
    ->schema([
        Placeholder::make('field1')->content(fn($state) => $state),
        Placeholder::make('field2')->content(fn($state) => $state),
        Placeholder::make('the rest')
            ->content(fn($state, $get) => // <-- how to get the full current item?
                collect($get('..')->except(['field1', 'field2'])
            ) 
    ])
```

**Comment by howdu** [comment]
~~I was just about to ask the same but it seems that `itemLabel` is the only thing that has access to the full repeater item.~~

As pointed out on [Twitter](https://x.com/filamentphp/status/1805107029259743568) you can do this with the component: `$component->getContainer()->getRawState()`

Another way is using hidden field:
```php
Tables\Actions\BulkAction::make('moreInfo')
    ->fillForm(fn (): array => [
        'items' => [
            [
                'reference' => 'AA',
                'name' => 'John Doe',
                'address' => '123 Main St.',
            ],
            [
                'reference' => 'AB',
                'name' => 'Jane Doe',
                'address' => '456 Elm St.',
            ]
        ]
    ])
    ->form([
        Repeater::make('items')
            ->itemLabel(fn (array $state): string => $state['reference'])
            ->schema([
                Hidden::make('address'),
                TextInput::make('name')
                    ->helperText(fn (Get $get): ?string => $get('address')),
            ]),
    ]);
  ```
